src/com/stanimiriliev/chaptertwo/homogeneoustree/Tree.java

```diff
@@ -1,94 +1,158 @@
+     */
+    public void printElements(){printElements(this.root);}
     /**
      * Prints the elements of the tree
      * First the root of the tree, then left nodes and right nodes
      */
-    public void printElements() {
-        System.out.println(this.value);// Prints the root element
-        if(this.left != null)// If bottom of the tree is not reached print left node
-            this.left.printElements();
-        if(this.right != null)// If bottom of the tree is not reached print right node
-            this.right.printElements();
+    public void printElements(Node root) {
+        System.out.println(root.value);
+        if(root.left != null)
+            printElements(root.getLeft());
+        if(root.right != null)
+            printElements(root.getRight());
     }
-
     /**
-     * Finds if the tree contains this element
-     * @param element -> The element to search for
-     * @return true if the element is found and false if not
+     * Inner class for Tree
      */
-    public boolean contains(int element) {
-        if(element == this.value)// If the root element is the searched element return true
-            return true;
-        else if(element < this.value) {// If the searched element is less than root search in left node
-            if(this.left != null) {// If the program is not reached the tree bottom recursively call this function with
-                // left sub node
-                return this.left.contains(element);
-            }
-            else// If the bottom is reached and the element is not found then it is not in the tree
-                return false;
+    class Node {
+        private final int value;
+        public Node left;
+        private Node right;
+
+        /**
+         * Constructor for new instances of this class
+         * @param value -> root of the tree
+         */
+        public Node(int value){
+            this(value, null, null);
         }
-        else {// If the searched element is greater than root search in right node
-            if(this.right != null) {// If the program is not reached the tree bottom recursively call this function with
-                // right sub node
-                return this.right.contains(element);
-            }
-            else// If the bottom is reached and the element is not found then it is not in the tree
-                return false;
+
+        /**
+         * Constructor for new instances of this class
+         * @param value -> root of the tree
+         * @param left -> left node of the tree
+         * @param right -> right node of the tree
+         */
+        public Node(int value, Node left, Node right){
+            this.value = value;
+            this.left = left;
+            this.right = right;
         }
+
+        /**
+         * Getter for the root of the tree
+         * @return the root
+         */
+        public int getValue(){return this.value;}
+
+        /**
+         * Getter for the left node of the tree
```

Comment:
do not write comment of such a methods, too much time.
And yes it will be better if you delete them and use directly the fields.
also the value is public final so you cannot nobody can change it for the node and you can access it directly.
Less code, cleaner solution(most of the times)